android/src/main/kotlin/com/rudderstack/sdk/kotlin/android/plugins/devicemode/DestinationPlugin.kt

```diff
@@ -0,0 +1,133 @@
+package com.rudderstack.sdk.kotlin.android.plugins.devicemode
+
+import com.rudderstack.sdk.kotlin.android.Configuration
+import com.rudderstack.sdk.kotlin.android.plugins.devicemode.eventprocessing.EventFilteringPlugin
+import com.rudderstack.sdk.kotlin.android.plugins.devicemode.eventprocessing.IntegrationOptionsPlugin
+import com.rudderstack.sdk.kotlin.android.utils.isFalseOrNull
+import com.rudderstack.sdk.kotlin.core.Analytics
+import com.rudderstack.sdk.kotlin.core.internals.logger.LoggerAnalytics
+import com.rudderstack.sdk.kotlin.core.internals.models.AliasEvent
+import com.rudderstack.sdk.kotlin.core.internals.models.Destination
+import com.rudderstack.sdk.kotlin.core.internals.models.Event
+import com.rudderstack.sdk.kotlin.core.internals.models.GroupEvent
+import com.rudderstack.sdk.kotlin.core.internals.models.IdentifyEvent
+import com.rudderstack.sdk.kotlin.core.internals.models.ScreenEvent
+import com.rudderstack.sdk.kotlin.core.internals.models.SourceConfig
+import com.rudderstack.sdk.kotlin.core.internals.models.TrackEvent
+import com.rudderstack.sdk.kotlin.core.internals.plugins.Plugin
+import com.rudderstack.sdk.kotlin.core.internals.plugins.PluginChain
+import kotlinx.serialization.json.JsonObject
+
+@Suppress("TooManyFunctions")
+abstract class DestinationPlugin : Plugin {
+
+    final override val pluginType: Plugin.PluginType = Plugin.PluginType.Destination
+
+    final override lateinit var analytics: Analytics
+
+    abstract val key: String
+    var isDestinationReady: Boolean = false
+        private set
+
+    private var isDestinationDisabledInSource: Boolean = false
+
+    private lateinit var pluginChain: PluginChain
+    private val pluginList: MutableList<Plugin> = mutableListOf()
```

Comment:
Let's use `destination` terminology to signify the actual use case:
```suggestion
    private val destinationPluginList: MutableList<Plugin> = mutableListOf()
```

Reply:
This is not really a list of destination plugins, but actually a list of generic plugins, hence the naming is appropriate.

Reply:
I'm sorry, I somehow forgot to delete this comment.

Clarification: I didn't mean to suggest that change in this class. Please ignore it.